pallets/asset-registry/src/lib.rs

```diff
@@ -748,15 +709,40 @@ impl<T: Config> Create<T::AssetNativeLocation, Balance> for Pallet<T> {
 
 		Self::do_register_asset(asset_id, &details, location)
 	}
-}
 
-use hydradx_traits::InspectRegistry;
-impl<T: Config> InspectRegistry<T::AssetId> for Pallet<T> {
-	fn exists(asset_id: T::AssetId) -> bool {
-		Assets::<T>::contains_key(asset_id)
-	}
+	fn get_or_register_asset(
+		name: &[u8],
+		kind: AssetKind,
+		existential_deposit: Option<Balance>,
+		symbol: Option<&[u8]>,
+		decimals: Option<u8>,
+		location: Option<Self::Location>,
+		xcm_rate_limit: Option<Balance>,
+		is_sufficient: bool,
+	) -> Result<Self::AssetId, Self::Error> {
+		//NOTE: in this case `try_into_bounded()` should never return None.
+		let bounded_name = match Self::try_into_bounded(Some(name.to_vec()))? {
+			Some(n) => n,
+			None => return Err(Error::<T>::InvalidAssetname.into()),
+		};
```

Comment:
```suggestion
		let bounded_name = Self::try_into_bounded(Some(name.to_vec())).map_err(|_| Error::<T>::InvalidAssetname)?;
```
Shouldn't need the match, no?

Reply:
But this would suppress error returned from `try_into_bounded()`. 
`InvalidAssetname` is something that should never happen. `try_into_bounded` returns `None` only if `None` was provided as param.